### PR TITLE
preliminary slice plugin

### PR DESCRIPTION
**Pull request by traeak**
A working implementation of the nginx slice plugin.  Needs more code review.

### CODE REVIEW

**Comment by traeak** [comment]
Couple of bugs:  on first block from server if response is 206 but Content-Range can't be parsed the client connection is just shut down.
For -<last bytes> request with more than one block I believe the first block skipbytes aren't correct (need to double check this).

**Comment by scw00** [comment]
[approve ci]

**Comment by scw00** [comment]
Thanks for your plugin.  I think there are some cases should be considered.
 - What should we do if client send the PURGE request for a cache object.
 - What should we do if cache object are expired.
 - What should we do if OS doesn't support range (you may already implemented this feature)

Note: we already have the clang-format file. You can run "make clang-format" to format your code. and  
you can write makefile.am instead your makefile.

**Comment by scw00** [comment]
Please squash your commits when this pr is ready to launch.

**Comment by traeak** [comment]
I accidentally left a bad .clang-format file in the plugin directory that was picked up.

Any non 206 response from a server will get passed back and not handled.  I believe a 200 is typical in these situations.

The above bugs could be considered as rare corner cases and aren't too difficult to fix so I think the plugin is ready for preview testing.

Cache control apparently is already a headache in ATS.  This one I'll have to defer for the time being and have this be prioritized hopefully with a solid proposed workflow.

From my testing with a test edge/real parent cache/real origin the results are very promising with origin timeouts and dropouts (during the big level3 outage today) being reasonably handled.  There has already been extensive local stress testing running days with 10's of TB through a local ats and apache server instance.  Some detailed functional testing will be done in the future, I want to simulate some very controlled origin/parent cache failures.

**Comment by scw00** [comment]
[approve ci]

**Comment by scw00** [comment]
Also remembering document your plugin :-).

File: plugins/experimental/slice/slice.cc [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+  return TS_SUCCESS;
+}
+
+///// global plugin
+SLICE_EXPORT
+void TSPluginInit(int argc, char const *argv[])
+{
+  TSPluginRegistrationInfo info;
+  info.plugin_name = (char *)PLUGIN_NAME;
+  info.vendor_name = (char *)"Comcast";
```
Comment (ezelkow1) [inline plugins/experimental/slice/slice.cc]: Dont forget to change these to apache

**Comment by traeak** [comment]
Made the 416 handling consistent with nginx.  Cache invalidation is handled via a different plugin that is run globally, I forget which one offhand

**Comment by traeak** [comment]
Hopefully the last fix, allowing the current slice block in progress to drain so that it is cached.

**Comment by zwoop** [comment]
Why was this closed? Is there another PR coming for this plugin ?

**Pull request closed without merging**